DctrWho/Controllers/DoctorController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using DoctorWho.Db.Interface;
+using DoctorWho.Dto;
+using DoctorWho.helper;
+using EfDoctorWho;
+using Microsoft.AspNetCore.Mvc;
+using AutoMapper;
+using EFCore;
+
+namespace DoctorWho.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class DoctorController : Controller
+    {
+        private readonly IDoctor _DoctorRepositry;
+        private readonly  IMapper _mapper;
+        public DoctorController(IDoctor DoctorRepositry, IMapper mapper )
+        {
+            _DoctorRepositry = DoctorRepositry;
+            _mapper = mapper;
+        }
+
+        [HttpGet("Doctors/", Name = "GetAllDoctors")]
+        [ProducesResponseType(200, Type = typeof(IEnumerable<Doctor>))]
+        public IActionResult GetAllDoctors()
+        {
+            var Doctors = _mapper.Map<List<DoctorDto>>(_DoctorRepositry.GetAllDoctors());
+
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            return Ok(Doctors);
+        }
+
+        [HttpDelete("Doctors/", Name = "DeleteDoctors")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(204)]
+        [ProducesResponseType(404)]
+        public IActionResult DeleteDoctors(int id)
+        {
+            if (_DoctorRepositry.DeleteDoctor(id))
+            {
+                return Ok();    
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+                 return NotFound();
+        }
+
+        [HttpPost("Doctors/",Name = "UpsertDoctors")]
+        public DoctorDto UpsertDoctors([FromBody] DoctorDto doctor)
+        {
+            var Doctors = _mapper.Map<List<DoctorDto>>(_DoctorRepositry.GetAllDoctors()).Find(s=>s.Id == doctor.Id);
```

Comment:
@noor848 another thing here, My expectation from Upsert is to create or update, so I expect to have a Doctor object without id since the id is Database auto-generated.
what you need to do here is to have the doctorNumber & Name as your key to check if it is exist